Python/pylifecycle.c

```diff
@@ -1835,8 +1836,16 @@ Py_FinalizeEx(void)
      * Alas, a lot of stuff may still be alive now that will be cleaned
      * up later.
      */
+
+    FILE *fp = stderr;
+    if (dump_file) {
+        char dump_file_name[255];
+        time_t now = time(NULL);
+        sprintf(dump_file_name, "cpython-tracerefs-%ld.dump", now);
```

Comment:
I suggest to use the pid rather than the timestamp. An alternative is that the environment variable contains the filename, so the user handles filename conflict. The problem is that sometimes, the correct directory is removed once the process completes. It is the case when a regrtest worker process completes.

PYTHONDUMPFILE=/path/to/dump.txt

Reply:
Pid looks better, I would to use PYTHONDUMPDIR instead of PYTHONDUMPFILE.
So the file will be created as `PYTHONDUMPFILE/cpython-tracerefs-<:pid>.txt`